BTMLColorLOSMod/WeaponRangeIndicators_DrawLine_Patch.cs

```diff
@@ -189,16 +189,24 @@ public static class WeaponRangeIndicators_DrawLine_Patch
                         {
                             // other than formatting this block is the only thing that changed from the decompiled code
                             Vector3[] pointsForArc = WeaponRangeIndicators.GetPointsForArc(18, 30f, vector, vector2);
-                            float shotQuality = (float) ReflectionHelper.InvokePrivateMethode(__instance,
-                                "GetShotQuality", new object[] {selectedActor, position, rotation, target});
-                            // alright future me, this is probably destructive in some way, but 
-                            // this lets us set the color of the line via that color6 bit.
-                            line.material.color = Color.white;
-                            Color color6 = Color.Lerp(
-                                Color.clear,
-                                BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcColor, 
-                                shotQuality);
-                            line.endColor = (line.startColor = color6);
+                            if (BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcDashed)
+                            {
+                                line.material = __instance.MaterialOutOfRange;
+                                line.material.color = BTMLColorLOSMod.ModSettings.IndirectLineOfFireArcColor;
```

Comment:
We probably don't want to lose the shotQuality bit, so maybe something more like pseudocode:
```
if dashed
  set new materials
else
  just set white material color
do the normal lerp'ing
```
If the line color setting messes with the display of the dashed line color, maybe that is not the thing to do. THoughts?

Reply:
5m :) I wasn't sure how you felt about that so I didn't want to touch it. Refactoring.